vscode/src/extension.ts

```diff
@@ -1,45 +1,93 @@
-// The module 'vscode' contains the VS Code extensibility API
-// Import the module and reference it with the alias vscode in your code below
 import * as vscode from "vscode";
-import { VsCodeExtension } from "./VsCodeExtension";
+import { KonveyorGUIWebviewViewProvider } from "./KonveyorGUIWebviewViewProvider";
+import { registerAllCommands } from "./commands";
+import { ExtensionState, SharedState } from "./extensionState";
+import { ViolationCodeActionProvider } from "./ViolationCodeActionProvider";
 import { AnalyzerClient } from "./client/analyzerClient";
 
-let client: AnalyzerClient;
+class VsCodeExtension {
+  private state: ExtensionState;
 
-// This method is called when your extension is activated
-// Your extension is activated the very first time the command is executed
-export function activate(context: vscode.ExtensionContext) {
-  // TODO(djzager): This was in continue but I couldn't get it to work correctly.
-  // const { activateExtension } = await import("./activate");
-  try {
-    const ext = new VsCodeExtension(context);
-    client = ext.getAnalyzerClient();
-
-    console.log("Extension activated");
-  } catch (e) {
-    console.log("Error activating extension: ", e);
-    vscode.window
-      .showInformationMessage(
-        "Error activating the Konveyor extension.",
-        //   "View Logs",
-        "Retry",
-      )
-      .then((selection) => {
-        //   if (selection === "View Logs") {
-        // 	vscode.commands.executeCommand("konveyor.viewLogs");
-        //   } else
-        if (selection === "Retry") {
-          // Reload VS Code window
-          vscode.commands.executeCommand("workbench.action.reloadWindow");
-        }
-      });
+  constructor(context: vscode.ExtensionContext) {
+    this.state = {
+      analyzerClient: new AnalyzerClient(context),
+      sharedState: new SharedState(),
+      webviewProviders: new Set<KonveyorGUIWebviewViewProvider>(),
+      sidebarProvider: undefined as any,
+      extensionContext: context,
+    };
```

Comment:
I don't know much about how vscode extensions handle things -- is this the normal way to handle state?

I wonder if using https://immerjs.github.io/immer/ would make any kind of sense.